packages/backend/src/core/SyncScheduler.ts

```diff
@@ -23,50 +26,64 @@ export class SyncScheduler {
   )
 
   async start() {
-    let last = await this.statusRepository.getLastBlockNumberSynced()
+    let lastSynced = await this.statusRepository.getLastBlockNumberSynced()
+    await this.dataSyncService.discardAfter(lastSynced)
 
-    this.logger.info('Last block number synced', { last })
+    const lastKnown = this.blockDownloader.getLastKnownBlock()
 
-    await this.dataSyncService.revert(last)
+    this.logger.info('start', { lastSynced, lastKnown: lastKnown.number })
 
-    this.safeBlockService.onNewSafeBlock(({ blockNumber }) => {
-      if (blockNumber > last) {
-        this.schedule({ from: last + 1, to: blockNumber })
-        last = blockNumber
-      }
+    if (lastKnown.number > lastSynced) {
+      const blocks = await this.blockRepository.getAllInRange(
+        lastSynced + 1,
+        lastKnown.number
+      )
+      lastSynced = lastKnown.number
+      this.scheduleSync(new BlockRange(blocks))
+    }
+
+    this.blockDownloader.onNewBlocks(async (newBlockRange) => {
```

Comment:
Isn't there a race condition here that between getting the last known block and subscribing the last known block actually changes and we miss some blocks?